### ISSUE TITLE
Collapse pool / contents erased

### ISSUE THREAD

**Issue by WalkerWest** [camunda/camunda-modeler#509]
Hey ... when I choose the "Collapsed Pool" option in 1.6.0 or the latest nightly build running on 64-bit Windows, and then choose "Expanded Pool" the pool contents are erased.  On the XML tab, the corresponding lines also disappear.

**Comment by nikku** [comment]
That is correct. There does not exist a concept like "Collapsed Pool with contents" in BPMN.

**Comment by nikku** [comment]
Reopening, as we could technically build something like this (working around BPMN limitations). See https://github.com/bpmn-io/bpmn-js/issues/659 for the workaround.

**Comment by felix-mueller** [comment]
@nikku is this still a won't fix or do you think at some point it will be on the roadmap?

**Comment by nikku** [comment]
This issue is not on our roadmap. Would be interesting to learn in which scenarios you think that functionality makes sense and how often you'd use it.

**Comment by ragnarnevries** [comment]
Got this question (and then slander) in three consecutive missions. Use case: model a huge collaboration for a complete process, then, for a department or the process engineer, collapse all pools except the interesting one while **keeping** the message flows. Even better if it is expandable afterwards.
Minimum requirement: Make the user clear that she looses the content by switching from expanded to collapsed pool.

**Comment by nikku** [comment]
> Make clear to the user that she looses the content and message flows by switching from expanded to collapsed pool.

How would you like to make that _clear_ to the user? It is due to the BPMN that collapsing (and keeping contents) is not possible.

**Comment by ragnarnevries** [comment]
>> Make clear to the user that she looses the content and message flows by switching from expanded to collapsed pool.
> How would you like to make that clear to the user? It is due to the BPMN that collapsing (and keeping contents) is not possible.

It's definitely a question of UI/UX. We have to prevent the frustration that arises when a user expands a previously collapsed pool and then notices that it is empty. Then it is to late to press CTRL-Z. Hence, we have to warn the user. I don't no, hammer a warning on the screen or whatever when a user changes an expanded pool to a collapsed pool.

Anyway, this is no solution. Since collapsing subprocesses without loosing content works, it is very unintuitive that it does not work for pools (I know the technical issue, just talking about the UX).

**Comment by nikku** [comment]
100% agreed! If BPMN 3.0 is ever getting released I hope for that to be fixed.

**Comment by ingorichtsmeier** [comment]
There is a chance to fix it in the camunda modeler in BPMN2.0: 

Move the content of the pool with a different namespace into an extension element of the process if you collapse it. On expanding, map them back to the process level.

**Comment by nikku** [comment]
Closing this as won't fix, as the current behavior is en par with the BPMN standard.

Based on your feedback I've created https://github.com/camunda/camunda-modeler/issues/712 to track the _would be awesome to have_ version of an intuitive collapse / expand participant behavior. 

Find the full specification of what we'd like to have in https://github.com/bpmn-io/bpmn-js/issues/736. If someone wants to start working on the topic or file a pull request --- I'm happy to share my thoughts on the implementation.
